library/entropy.c

```diff
@@ -104,7 +104,11 @@ void mbedtls_entropy_init( mbedtls_entropy_context *ctx )
                                 MBEDTLS_ENTROPY_MIN_HAVEGE,
                                 MBEDTLS_ENTROPY_SOURCE_STRONG );
 #endif
-#if defined(MBEDTLS_ENTROPY_HARDWARE_ALT)
+#if defined(DEVICE_TRNG)
```

Comment:
This is an mbed OS define. I am not sure we would like to include something like this in mbed TLS. Another idea is to use `#if defined(MBEDTLS_ENTROPY_MIN_HARDWARE_ALT)`.